From d978929db7405b5e584f827ad25abd9a4b440cde Mon Sep 17 00:00:00 2001
From: Radu Pirea <radu-nicolae.pirea@nxp.com>
Date: Wed, 16 Dec 2020 16:30:08 +0200
Subject: [PATCH 14/76] campps32v2: reduce ddr speed to 400MHz

Reduce ddr speed to 400MHz for SoCs 1, 2 and 6 because are unstable at
533 MHz.

Issue: ALB-6158

Signed-off-by: Radu Pirea <radu-nicolae.pirea@nxp.com>
---
 board/freescale/campps32v2/lpddr2_config.c | 54 +++++++++++-----------
 1 file changed, 27 insertions(+), 27 deletions(-)

diff --git a/board/freescale/campps32v2/lpddr2_config.c b/board/freescale/campps32v2/lpddr2_config.c
index e6fc629903..577a9a6c0e 100644
--- a/board/freescale/campps32v2/lpddr2_config.c
+++ b/board/freescale/campps32v2/lpddr2_config.c
@@ -10,17 +10,17 @@
 static const struct lpddr2_config campps32v2_lpddr2_config_1 = {
 	.mdasp_module0 = 0x0000004F,
 	.mdasp_module1 = 0x0000006F,
-	.mdcfg0 = 0x454F61A5,
-	.mdcfg1 = 0x93F60EA3,
-	.mdcfg2 = 0x000000DD,
-	.mdcfg3lp = 0x001F0999,
+	.mdcfg0 = 0x33374133,
+	.mdcfg1 = 0xDAF00A82,
+	.mdcfg2 = 0x00000093,
+	.mdcfg3lp = 0x00170777,
 	.mdctl = 0x03110000,
 	.mdmisc = 0x00001688,
-	.mdscr_mr2 = 0x06028030,
-	.mprddlctl_module0 = 0x46484646,
-	.mprddlctl_module1 = 0x42464042,
-	.mpwrdlctl_module0 = 0x46484240,
-	.mpwrdlctl_module1 = 0x3C3E403E,
+	.mdscr_mr2 = 0x04028030,
+	.mprddlctl_module0 = 0x44444242,
+	.mprddlctl_module1 = 0x44484440,
+	.mpwrdlctl_module0 = 0x44444044,
+	.mpwrdlctl_module1 = 0x3E3C3A3E,
 };
 
 static const struct lpddr2_config campps32v2_lpddr2_config_2 = {
@@ -74,33 +74,33 @@ static const struct lpddr2_config campps32v2_lpddr2_config_4 = {
 static const struct lpddr2_config campps32v2_lpddr2_config_5 = {
 	.mdasp_module0 = 0x0000004F,
 	.mdasp_module1 = 0x0000006F,
-	.mdcfg0 = 0x454F61A5,
-	.mdcfg1 = 0x93F60EA3,
-	.mdcfg2 = 0x000000DD,
-	.mdcfg3lp = 0x001F0999,
+	.mdcfg0 = 0x33374133,
+	.mdcfg1 = 0xDAF00A82,
+	.mdcfg2 = 0x00000093,
+	.mdcfg3lp = 0x00170777,
 	.mdctl = 0x03110000,
 	.mdmisc = 0x00001688,
-	.mdscr_mr2 = 0x06028030,
-	.mprddlctl_module0 = 0x46464244,
-	.mprddlctl_module1 = 0x42463E3E,
-	.mpwrdlctl_module0 = 0x484A4846,
-	.mpwrdlctl_module1 = 0x3C3C3C3E,
+	.mdscr_mr2 = 0x04028030,
+	.mprddlctl_module0 = 0x42443C42,
+	.mprddlctl_module1 = 0x44444040,
+	.mpwrdlctl_module0 = 0x42444240,
+	.mpwrdlctl_module1 = 0x3E3E3E3C,
 };
 
 static const struct lpddr2_config campps32v2_lpddr2_config_6 = {
 	.mdasp_module0 = 0x0000004F,
 	.mdasp_module1 = 0x0000006F,
-	.mdcfg0 = 0x454F61A5,
-	.mdcfg1 = 0x93F60EA3,
-	.mdcfg2 = 0x000000DD,
-	.mdcfg3lp = 0x001F0999,
+	.mdcfg0 = 0x33374133,
+	.mdcfg1 = 0xDAF00A82,
+	.mdcfg2 = 0x00000093,
+	.mdcfg3lp = 0x00170777,
 	.mdctl = 0x03110000,
 	.mdmisc = 0x00001688,
-	.mdscr_mr2 = 0x06028030,
-	.mprddlctl_module0 = 0x46484848,
-	.mprddlctl_module1 = 0x484A4444,
-	.mpwrdlctl_module0 = 0x46464240,
-	.mpwrdlctl_module1 = 0x3C3C3C3E,
+	.mdscr_mr2 = 0x04028030,
+	.mprddlctl_module0 = 0x44443E44,
+	.mprddlctl_module1 = 0x46484040,
+	.mpwrdlctl_module0 = 0x46424240,
+	.mpwrdlctl_module1 = 0x3C3C3E3C,
 };
 
 const struct lpddr2_config *s32_get_lpddr2_config(void)
-- 
2.17.1

